### PR TITLE
BufferStream: Add signed int overloads for read/write

### PR DESCRIPTION
**Pull request by rburchell**
This will be required in AudioServer

### CODE REVIEW

File: AK/BufferStream.h [hunk 1]
```diff
@@ -126,6 +187,12 @@ class BufferStream {
         m_buffer.trim(m_offset);
     }
 
+    void reset()
```
Comment (rburchell) [inline AK/BufferStream.h]: Oh, I suppose this should be a separate commit. And I should probably fix the = 0 -> = false :)

**Comment by awesomekling** [comment]
Looks great! :)
At some point we should probably consider if we don't want the `<<` operators to also do overflow checking...
